src/vectorclock.erl

```diff
@@ -124,91 +124,33 @@ set_clock_of_dc(DcId, Time, VectorClock) ->
 from_list(List) ->
     dict:from_list(List).
 
-eq(V1, V2) ->
-    dict:fold( fun(Dcid, Time2, Result) ->
-                       case dict:find(Dcid, V1) of
-                           {ok, Time1} ->
-                               case Time1 =:= Time2 of
-                                   true ->
-                                       Result;
-                                   false ->
-                                       false
-                               end;
-                           error ->
-                               false
-                       end
-               end,
-               true, V2).
+-spec for_all_keys(fun((non_neg_integer(), non_neg_integer()) -> boolean()), vectorclock(), vectorclock()) -> boolean().
+for_all_keys(F, V1, V2) ->
+  %% We could but do not care about duplicate DC keys - finding duplicates is not worth the effort
```

Comment:
Vectorclock is a dict mapping from a DCID to a timestamp. If a given DCID key is absent, the default value is zero. When comparing two vectorclocks, you need to run the comparison predicate (<, >, == etc) for each key in both vectorclocks. We could compute a union of two sets of keys (complexity n log n), but the faster way is to just run the same predicate for all keys in the first dict, and then run the same predicate for the keys in second dict (complexity n), assuming that the predicate itself is of complexity 1.

We will compare the same entries twice, but in the end it is still faster.